Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py

```diff
@@ -173,6 +173,33 @@ def password_change_user(
         }
         self.ms_client.http_request(method="PATCH", url_suffix=f"users/{quote(user)}", json_data=body, resp_type="text")
 
+    def fetch_password_method_id(self, user: str) -> str:
+        """
+        fetches the password method ID, to be used later. See API docs for reference.
+        """
+        password_method_id_response = None
+        try:
+            password_method_id_response = self.ms_client.http_request(
+                method="GET", url_suffix=f"users/{quote(user)}/authentication/passwordMethods"
+            )
+            password_method_id = password_method_id_response.get("value", [])[0]["id"]
+        except (IndexError, KeyError) as e:
+            raise DemistoException("Failed getting passwordMethod id", exception=e, res=password_method_id_response)
+        return password_method_id
+
+    def password_change_user_on_premise(self, user: str, password: str, password_method_id: str) -> None:
+        """
+        changes the password of a user on premise.
+        """
+        self.ms_client.http_request(
+            method="POST",
+            url_suffix=f"users/{quote(user)}/authentication/passwordMethods/{password_method_id}/resetPassword",
+            ok_codes=(202,),
+            json_data={"newPassword": password},
+            return_empty_response=True,
+            resp_type=None,
```

Comment:
Not sure that you need here the response type.

Reply:
I added
`empty_valid_codes=(202,)`
and removed 
`resp_type=None`

In the CSP code, it will be used:
<img width="651" height="115" alt="image" src="https://github.com/user-attachments/assets/77445d9a-cb26-4fd3-adb6-998b245ac667" />